### ISSUE TITLE
Submarine submerge and retreat issues

### ISSUE THREAD

**Issue by panther2** [triplea-game/triplea#4821]
### Engine version
Tested with 1.10.14614

I am very short of time at the moment so this is more or less an announcement of issues that have to be deeper investigated for all versions. While testing https://github.com/triplea-game/triplea/pull/4820 and taking a quick glance on some other versions I noticed this:

First issue:
When the attacker chooses to submerge TripleA alway submerges **all** attacking submarines.
Instead the player should be able to choose how many of his subs should submerge.
Also the defender should be able to select how many of his subs should submerge,

Second issue:
When the attacker chooses to retreat submerged subs are retreated together with the surface ships. Instead the submerged ships should remain in the contested seazone.

I am sorry for this "unprofessional" report, but as I said, I don't have much time. So any help in investigating and testing is highly welcome.

CC: @ron-murhammer

**Comment by ron-murhammer** [comment]
@panther2 Interesting. I didn't realize you could choose to submerge only specific subs. I figured it was like retreating units which is all or nothing. Are there certain scenarios you can think of that you'd want to submerge only some subs?

**Comment by panther2** [comment]
@ron-murhammer Yes there are, Sometimes you want to leave (one or more) submerged submarines in the contested seazone. When they resurface the opponents have to deal with them in any way. 
At the same time you maybe want to retreat your (non submerged) subs together with the fleet to an adjacent space.

Older rulesets state the decision on a single submarine basis. Newer rulesets explicitly state "submerge any number of subs". No ruleset says "all subs". 
Also submerged subs should not been moved with the retreating fleet.

When I have some more time, I will check the engine map version by map version (ruleset by ruleset) concerning those issues. I just can't do it these days.

**Comment by trevan** [comment]
I have a fix for the second issue in #7007
